burn-slashed-collateral.md

```diff
@@ -0,0 +1,123 @@
+    liars_count: usize,
+    // To calculate the reward, we consider errors_count as the number of errors that are
+    // out of consensus, it means, that they do not deserve any reward
+    errors_count: usize,
+    reward: u64,
+    collateral: u64,
+) -> (u64, u64) {
+    let honests_count = (commits_count - liars_count - errors_count) as u64;
+
+    if commits_count == 0 {
+        (0, 0)
+    } else if honests_count == 0 {
+        (collateral, 0)
+    } else {
+        let burned_collateral = collateral * (liars_count as u64);
+
+        (
+            reward + collateral,
+            burned_collateral,
+        )
+    }
+}
+```
+
+Furthermore, the `create_tally` function will need to add an extra output to the transaction if it contains any liars. This output could take similar form as the `change_output`:
+
+```
+let (reward, burned) = calculate_witness_reward(
+    commits_count,
+    liars_count + non_reveals_count,
+    errors_count,
+    dr_output.witness_reward,
+    collateral,
+)
+
+...
+
+if liars_count > 0 {
+    let vt_output_burn = ValueTransferOutput {
+        wit10000000000000burnbabyburn0000000000000,
+        value: burned,
+        time_lock: 0,
+    };
+    outputs.push(vt_output_burn);
+}
+
+...
+
+```
+
+## Backwards compatibility
+
+- Implementer: a client that implements or adopts the protocol improvements proposed in this document.
+- Non-implementer: a client that fails to or refuses to implement the protocol improvements proposed in this document.
+
+#### Consensus cheat sheet
+
+Upon entry into force of the proposed improvements:
+
+- Blocks and transactions that were considered valid by former versions of the protocol MUST continue to be considered valid.
+- Blocks and transactions produced by non-implementers MUST be validated by implementers using the same rules as with those coming from implementers, that is, the new rules.
+- Implementers MUST apply the proposed logic when evaluating transactions and proposed blocks.
+
+As a result:
+
+- Non-implementers MAY NOT get their blocks and transactions accepted by implementers.
+- Implementers MAY get their valid blocks accepted by non-implementers.
+- Non-implementers MAY consolidate different blocks and superblocks than implementers.
+
+Due to this last point, this MUST be considered a consensus-critical protocol improvement. An adoption plan MUST be proposed, setting an activation date that gives miners enough time in advance to upgrade their existing clients.
+
+## Reference Implementation
+
+A reference implementation for the proposed protocol improvement can be found in the [pull request #2229](https://github.com/witnet/witnet-rust/pull/2229) of the [witnet-rust] repository.
+
+## Adoption Plan
+
+An activation date is proposed for <Month> <Day> <Year> at 9am UTC, that is, protocol epoch #<Epoch>.
+
+The signaling bit for this improvement proposal is `4` (`0b00000000000000000000000000010000`) and the `first_signaling_epoch` is set to #<Epoch>.
```

Comment:
Drafts should skip the adoption plan or just state that a proper adoption plan will be provided later as the proposal moves from Draft to Proposed.